src/dtx/dtx_common.c

```diff
@@ -1487,15 +1485,9 @@ dtx_leader_end(struct dtx_leader_handle *dlh, struct ds_cont_hdl *coh, int resul
 	/* If piggyback DTX has been done everywhere, then need to handle CoS cache.
 	 * It is harmless to keep some partially committed DTX entries in CoS cache.
 	 */
-	if (result == 0 && dth->dth_cos_done) {
```

Comment:
just confirm why this branch removed? thx

Reply:
The condition will checked via the parameter for dtx_cos_put_piggyback().